code/modules/surgery/implant.dm

```diff
@@ -1,228 +1,363 @@
-		var/obj/item/organ/external/chest/affected = target.get_organ(target_zone)
+	target.custom_pain("The pain in your [target_zone] is living hell!",1)
+	..()
 
-		if(istype(tool, /obj/item/weapon/disk/nuclear))
-			user << "Central command would kill you if you implanted the disk into someone."
+/datum/surgery_step/cavity/place_item/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool,datum/surgery/surgery)
+	var/obj/item/organ/external/chest/affected = target.get_organ(target_zone)
+
+	if(istype(tool, /obj/item/weapon/disk/nuclear))
+		user << "Central command would kill you if you implanted the disk into someone."
+		return 0//fail
+
+	if(istype(tool,/obj/item/organ))
+		user << "This isn't the type of surgery for that!"
+		return 0//fail
+
+	if(tool)
+		if(IC)
+			user << "There seems to be something in there already!"
+			return 1
 		else
-			user.visible_message("\blue [user] puts \the [tool] inside [target]'s [get_cavity(affected)] cavity.", \
-			"\blue You put \the [tool] inside [target]'s [get_cavity(affected)] cavity." )
-			if (tool.w_class > get_max_wclass(affected)/2 && prob(50) && !(affected.status & ORGAN_ROBOT))
-				user << "\red You tear some vessels trying to fit such big object in this cavity."
+			user.visible_message("<span class='notice'> [user] puts \the [tool] inside [target]'s [get_cavity(affected)] cavity.</span>", \
+			"<span class='notice'> You put \the [tool] inside [target]'s [get_cavity(affected)] cavity.</span>" )
+			if((tool.w_class > get_max_wclass(affected)/2 && prob(50) && !(affected.status & ORGAN_ROBOT)))
+				user << "<span class='warning'> You tear some vessels trying to fit the object in the cavity.</span>"
 				var/datum/wound/internal_bleeding/I = new ()
 				affected.wounds += I
 				affected.owner.custom_pain("You feel something rip in your [affected.name]!", 1)
 			user.drop_item()
 			affected.hidden = tool
+			target.internal_organs += tool
 			tool.loc = target
 			affected.cavity = 0
+			return 1
+	else
+		if(IC)
+			user.visible_message("[user] pulls [IC] out of [target]'s [target_zone]!", "<span class='notice'>You pull [IC] out of [target]'s [target_zone].</span>")
+			user.put_in_hands(IC)
+			target.internal_organs -= IC
+			return 1
+		else
+			user << "<span class='warning'>You don't find anything in [target]'s [target_zone].</span>"
+			return 0
+
 
 //////////////////////////////////////////////////////////////////
 //					IMPLANT/ITEM REMOVAL SURGERY						//
 //////////////////////////////////////////////////////////////////
 
+/datum/surgery/cavity_implant_rem
+	name = "implant removal"
+	steps = list(/datum/surgery_step/generic/cut_open, /datum/surgery_step/generic/clamp_bleeders, /datum/surgery_step/generic/retract_skin,/datum/surgery_step/cavity/implant_removal,/datum/surgery_step/cavity/close_space,/datum/surgery_step/generic/cauterize/)
+	possible_locs = list("chest","head")//head is for borers..i can put it elsewhere
```

Comment:
for this surgery, I don't think the `affected.hidden` section is needed anymore, correct? I believe that relates to cavity surgery, which is handled above, right? In that case, you wouldn't want to check for head or chest.

Likewise, you wouldn't need to check for external organs either, I believe.